lib/SILGen/SILGenPattern.cpp

```diff
@@ -2132,16 +2165,18 @@ void PatternMatchEmission::emitEnumElementDispatch(
         eltConsumption = CastConsumptionKind::TakeAlways;
       }
 
-      SILValue eltValue;
+      ManagedValue eltValue;
       // We can only project destructively from an address-only enum, so
       // copy the value if we can't consume it.
       // TODO: Should have a more efficient way to copy payload
       // nondestructively from an enum.
       switch (eltConsumption) {
-      case CastConsumptionKind::TakeAlways:
-        eltValue =
-            SGF.B.createUncheckedTakeEnumDataAddr(loc, srcValue, elt, eltTy);
+      case CastConsumptionKind::TakeAlways: {
+        auto finalValue = src.getFinalManagedValue();
```

Comment:
Is it valid to use a ManagedValue after having called forward() on it? Maybe we should stash the return value of the call to forward() above?

Reply:
I pointed out to Slava there is an `} else {` above. Resolving!